### PR TITLE
Hide SingleSelect clear button when disabled

### PR DESCRIPTION
**Pull request by gdostie**
The clear button was shown and was also effective even when the SingleSelect was disabled.

What I did:
- Prevented the clear button from showing when the SingleSelect is disabled
- Prevented the clear single select value action to be dispatched onclick when the SingleSelect is disabled

### CODE REVIEW

File: src/components/listBox/ListBox.tsx [hunk 1]
```diff
@@ -83,9 +84,14 @@ export class ListBox extends React.Component<IListBoxProps, {}> {
             />)
             .value();
 
-        return items.length
-            ? items
-            : <ItemBox {...{...this.props.noResultItem, classes: ['multi-line']}} />;
+        const emptyItem = (
```
Comment (pochretien) [inline src/components/listBox/ListBox.tsx]: We can allowed a ReactNode as props for the empty box I think, Its more flexible if we want a specific design not related to a ItemBox

File: src/components/select/SingleSelectConnected.tsx [hunk 2]
```diff
@@ -121,11 +121,17 @@ export class SingleSelectConnected extends React.Component<ISingleSelectProps &
...
     private getDeselectOptionButton(): React.ReactNode {
         return (
-            <Tooltip title={this.props.deselectTooltipText} placement='top' noSpanWrapper onClick={this.props.deselect}>
+            <Tooltip title={this.props.deselectTooltipText} placement='top' noSpanWrapper onClick={this.handleDeselect}>
                 <Svg svgName={VaporSVG.svg.clear.name} svgClass='icon mod-12' className='btn-append center-align' />
             </Tooltip>
         );
     }
+
+    private handleDeselect = () => {
```
Comment (pochretien) [inline src/components/select/SingleSelectConnected.tsx]: return your event to deselect if we do this

Comment (gdostie) [reply src/components/select/SingleSelectConnected.tsx]: The `deselect` prop isn't typed to accept an event as argument, and the event isn't used in the `mapDispatchToProps` anyway.

File: src/components/listBox/ListBox.tsx (hunk 1 above)
Comment (gdostie) [reply src/components/listBox/ListBox.tsx]: I like the idea, not sure if it is a breaking change though, I will try it.

Comment (gdostie) [reply src/components/listBox/ListBox.tsx]: It involves too much changes and its probably breaking, so I think its out of the scope of this fix. Would be nice to do though.

**Pull request merged**